### PR TITLE
Add new cli to add a interface ip as secondary address

### PR DESCRIPTION
**Pull request by shbalaku-microsoft**
#### What I did

Added a new CLI command to add secondary subnet value. 

#### How I did it

In the config class, added a snippet of code where the code to validate the secondary field and set the appropriate flag.

#### How to verify it

Added a test to validate the added flag

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -4551,6 +4552,11 @@ def add(ctx, interface_name, ip_addr, gw):
             config_db.set_entry(table_name, interface_name, {"NULL": "NULL"})
     config_db.set_entry(table_name, (interface_name, str(ip_address)), {"NULL": "NULL"})
 
+    if secondary:
+        # We update the secondary flag only in case of VLAN Interface.
+        if table_name == "VLAN_INTERFACE":
```
[the remark this answers is not in the document]
Comment (venkatmahalingam) [reply config/main.py:4557]: Do we need a validation to make sure not all IPs are marked as secondary?

[the remark this answers is not in the document]
Comment (venkatmahalingam) [reply config/main.py:4557]: Do we need a validation to make sure no more than one primary IP for a given VLAN interface?

[the remark this answers is not in the document]
Comment (pushpraj) [reply config/main.py:4557]: > Do we need a validation to make sure not all IPs are marked as secondary?

Ack, this is a good check.


> Do we need a validation to make sure no more than one primary IP for a given VLAN interface?

This we cannot do for two reasons: a) impact to potential existing devices with more than one interface b) interchanging primary with secondary would need more than one primary intermittently.

File: tests/ip_config_test.py [hunk 2]
```diff
@@ -84,6 +84,18 @@ def test_add_del_interface_valid_ipv4(self):
         assert result.exit_code != 0
         assert ('Eth36.10', '32.11.10.1/24') not in db.cfgdb.get_table('VLAN_SUB_INTERFACE')
 
+        # config int ip add Ethernet0.10 10.21.20.1/24 as secondary
+        result = runner.invoke(config.config.commands["interface"].commands["ip"].commands["add"], ["Vlan100", "10.11.20.1/24", "--secondary"], obj=obj)
+        assert result.exit_code == 0
+        assert ('Vlan100', '10.11.20.1/24') in db.cfgdb.get_table('VLAN_INTERFACE')
+        assert db.cfgdb.get_table('VLAN_INTERFACE')[('Vlan100', '10.11.20.1/24')]['secondary'] == "true"
```
Comment (pushpraj) [inline tests/ip_config_test.py]: nit: add a test where 'secondary' is false for non-secondary cli call.

File: config/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pushpraj) [reply config/main.py:4556]: This has been addressed. Only vlans with atleast one primary can be added with secondaries.

File: tests/ip_config_test.py (hunk 2 above)
Comment (shbalaku-microsoft) [reply tests/ip_config_test.py]: Added the test case for this scenario.

**Review by pushpraj — APPROVED** [review]

**Comment by shbalaku-microsoft** [comment]
@venkatmahalingam can you review the PR?

**Comment by zhangyanzhao** [comment]
@shbalaku-microsoft  please make sure the new CLI is updated to https://github.com/sonic-net/sonic-utilities/blob/master/doc/Command-Reference.md

**Review by prsunny — DISMISSED** [review]

**Comment by prsunny** [comment]
@venkatmahalingam , can you signoff?

**Review by venkatmahalingam — APPROVED** [review]

**Comment by irene-pan1202** [comment]
@shbalaku-microsoft 
The design of the secondary IP seems to have an issue that hasn't been considered? 
1. If a VLAN interface has only an IPv6 address, it is also considered as the primary IP, and an IPv4 secondary IP can still be successfully configured.
2. Deleting the primary IP seems to not check if the secondary IP exists?

**Comment by zhangyanzhao** [comment]
@qiluo-msft  can we merge this PR? Thanks.

**Comment by qiluo-msft** [comment]
Please merge latest master branch. The semgrep checker is added after this PR raised, so it will always block.

**Comment by prsunny** [comment]
@shbalaku-microsoft , can you please rebase the code?

**Comment by zhangyanzhao** [comment]
@shbalaku-microsoft  kind reminder to rebase the code, so that we can merge this feature before 202405 fork date on 5/30

**Review by prsunny — APPROVED** [review]

**Pull request merged**